engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java

```diff
@@ -32,7 +32,7 @@
 public final class AtomConstructor implements TruffleObject {
 
   private final String name;
-  private @CompilerDirectives.CompilationFinal ModuleScope definitionScope;
+  private final @CompilerDirectives.CompilationFinal ModuleScope definitionScope;
```

Comment:
If you make it final then
```suggestion
  private final ModuleScope definitionScope;
```

Reply:
Yes, `final` supersedes `@CompilationFinal`.